File: week3/code/get_TreeHeight.py
import sys
import pandas as pd
import numpy as np
import os

argument = sys.argv[1]

TreeData = pd.read_csv(argument)

def tree_height(degrees, distance):
    """Function using trigonometry to work out tree height"""
    radians = np.deg2rad(degrees)
    height = distance * np.tan(radians)
    return height

TreeData['Height'] = tree_height(TreeData['Angle.degrees'], TreeData['Distance.m'])

output_name = os.path.splitext(os.path.basename(sys.argv[1]))[0]

output_path = os.path.join("../results", f"{output_name}_treeheights.csv")

TreeData.to_csv(output_path, index=False)

File: week3/code/get_TreeHeight.R
library(tools)
arguments = commandArgs(trailingOnly = TRUE)
# use the first argument!
TreeData <- read.csv(arguments[1])

# function working out height
TreeHeight <- function(degrees, distance) {
  radians <- degrees * pi / 180
  height <- distance * tan(radians)
  # print(paste("Tree height is:", height))
  return (height)
}

TreeData$Height <- TreeHeight(TreeData$Angle.degrees,TreeData$Distance.m)

# remove filepath and extension
output_name <- filename_without_path_and_ext <- file_path_sans_ext("../data/trees.csv")
output_name <- basename(output_name)

name <- paste("../results/",output_name,"_treeheights.csv", sep = "")
write.csv(TreeData,name)



